Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -365,6 +459,23 @@ const RenderBox* ViewTimeline::sourceScrollerRenderer() const
     return subjectRenderer->enclosingScrollableContainer();
 }
 
+const RenderElement* ViewTimeline::stickyContainer() const
+{
+    auto subject = m_subject.styleable();
+    if (!subject)
+        return nullptr;
+
+    CheckedPtr renderer = subject->renderer();
+
+    auto scrollerRenderer = sourceScrollerRenderer();
+    while (renderer && renderer != scrollerRenderer) {
+        if (renderer->isStickilyPositioned())
+            return renderer.get();
+        renderer = renderer->parent();
```

Comment:
Generally it's better to walk the containing block chain, since sticky forms a containing block. If `renderer` is a fixedpos, this code would be wrong.

Reply:
Fixed.